Source/WebCore/page/writing-tools/WritingToolsController.mm

```diff
@@ -731,18 +769,6 @@
         didEndWritingToolsSession<WritingTools::Session::Type::Composition>(accepted);
         break;
     }
-
-    auto sessionRange = activeSessionRange();
-    if (!sessionRange) {
-        ASSERT_NOT_REACHED();
-        return;
-    }
-
-    m_page->chrome().client().removeInitialTextAnimationForActiveWritingToolsSession();
```

Comment:
What cancels the initial animation if the user exits Writing Tools by clicking done?

Reply:
Once the animation finishes, the initial animation is removed via the call to `clearAnimationsForActiveWritingToolsSession`

Reply:
The initial animation can go on indefinitely though.

Reply:
specifically, I mean if the destination animation finishes (or, if it never starts, then it's considered "finished" immediately)